components/definition/stimulationSources/NetPyNEStimulationSource.js

```diff
@@ -64,12 +64,18 @@ export default class NetPyNEStimulationSource extends React.Component {
   };
 
   updateLayout() {
-    const getType = (value) => {
-      Utils
-        .sendPythonMessage("'" + value + "' in netParams.stimSourceParams['" + this.state.currentName + "']['type']")
-        .then((response) => { if (response == true) { this.setState({ sourceType: value }) } });
-    };
-    this.stimSourceTypeOptions.forEach((option) => { getType(option.type) });
+    var opts = this.stimSourceTypeOptions.map((option) => { return option.type });
+    Utils
+      .sendPythonMessage("[value == netParams.stimSourceParams['" + this.state.currentName + "']['type'] for value in "+JSON.stringify(opts)+"]")
+      .then((responses) => {
+        if (responses.constructor.name == "Array"){
```

Comment:
I am bit lost here... what are you actually changing?

Reply:
We have **['VClamp', 'IClamp', 'NetStim',  'AlphaSynapse', 'SEClamp']** as options for this menu. Instead of sending 5 python messages asking if a variable is equal to one of them, I send an unique python message which returns: 
**[False, False, False, True, False]**, so I know the 4th element is the one I need to pick.
The python message code translates into this:

for value in opts:
....if value == netParams.stimSourceParams[blabla][blabla]:
........return True
....else:
........return False

Reply:
Ok, makes sense... although it would be good to move it to a different branch ;) No worries this time